source/appModules/eclipse.py

```diff
@@ -17,14 +22,83 @@ def event_valueChange(self):
 		# Therefore, just drop this event.
 		pass
 
+class AutocompletionListView(IAccessible):
+
+	def _get_name(self):
+		return AUTOCOMPLETION_LIST_NAME
+
+class AutocompletionListItem(IAccessible):
+
+	def event_selection(self):
+		# Fixme: I don't know if it is the correct way to do this.
+
+		# I think that you can feel more confortable if the text editor gain the focus after the item becomes hidden.
+		if self.appModule.textEditor == None:
+			self.appModule.textEditor = api.getFocusObject()
+
+		if self.appModule.selectedItem != self:
+			self.appModule.selectedItem = self
+			api.setFocusObject(self)
+			self.reportFocus()
+
+	def script_returnToTextEditor(self, gesture):
+		# We need to send the gesture to hide the autocompletion window
+		gesture.send()
+
+		# Retrieve the focus to the editor
+		if self.appModule.textEditor:
+			api.setFocusObject(self.appModule.textEditor)
+			self.appModule.textEditor.reportFocus()
+			
+			# And prevent confusing events
+			self.appModule.textEditor = None
+		
+		# And clean up the selected list item because we don't need it
+		self.appModule.selectedItem = None
+
+	def script_readDocumentation(self, gesture):

```

Comment:
What does this script do? What documentation? The documentation for the identifier associated with this autocomplete item? This needs a translatable docstring which will be used in input help and in the Input Gestures dialog. See the ["Scripts and Gesture Bindings" section of the Developer Gide](http://www.nvaccess.org/files/nvda/documentation/developerGuide.html#toc17) for details.